orderer/consensus/smartbft/chain_test.go

```diff
@@ -0,0 +1,709 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package smartbft_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/SmartBFT-Go/consensus/pkg/types"
+	"github.com/SmartBFT-Go/consensus/smartbftprotos"
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/msp"
+	"github.com/hyperledger/fabric/bccsp"
+	"github.com/hyperledger/fabric/bccsp/sw"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/metrics/disabled"
+	"github.com/hyperledger/fabric/common/policies"
+	policiesMocks "github.com/hyperledger/fabric/common/policies/policies_mocks"
+	"github.com/hyperledger/fabric/orderer/common/cluster"
+	clusterMocks "github.com/hyperledger/fabric/orderer/common/cluster/mocks"
+	msgprocessorMocks "github.com/hyperledger/fabric/orderer/common/msgprocessor/msgprocessor_mocks"
+	"github.com/hyperledger/fabric/orderer/consensus/smartbft"
+	smartbftMocks "github.com/hyperledger/fabric/orderer/consensus/smartbft/mocks"
+	"github.com/hyperledger/fabric/protoutil"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+)
+
+const (
+	eventuallyTimeout = 10 * time.Second
+)
+
+var (
+	logger        = flogging.MustGetLogger("smartBFT.chain.test")
+	nonce  uint64 = 0
+)
+
+//////////////////////////////////////////////////////////////
+// Tests
+//////////////////////////////////////////////////////////////
+
+func TestBftChain(t *testing.T) {
+	g := NewWithT(t)
+	channelId := "channel-test"
+
+	t.Run("TestSubmitingTxThenRestartingAllThenSubmitingAnotherTxShouldAddBothTxToLedger", func(t *testing.T) {
```

Comment:
The test name can have spaces so it is human readable.
It would be beneficial to add a comment on top of each case:

// Scenario: 
// start x nodes, configured such and such
// do bla... expect bla bla..

because these tests are going to be complicated and numerous, better do a free text description to help the reader.